deps/first/DynamicOutput/include/DynamicOutput/FileDevice.hpp

```diff
@@ -58,7 +58,7 @@ namespace RC::Output
     {
         if (m_always_create_file)
         {
-            m_file = File::open(m_file_name_and_path, File::OpenFor::Appending, File::OverwriteExistingFile::Yes, File::CreateIfNonExistent::Yes);
+            m_file = File::open(m_file_name_and_path, File::OpenFor::Writing, File::OverwriteExistingFile::Yes, File::CreateIfNonExistent::Yes);
```

Comment:
We actually do want to append though.

Reply:
hmm.. in that case you should just use `m_always_create_file=false`, which will append also create if not exists..
Append + OverwriteExisting makes no sense.

Reply:
The `Append` happens when the file gets written to, and the `OverwriteExisting` happens when `File` is constructed via `File::open`.
We want to overwrite the file if it exists, and after that we want to append to the file every time anything is written to it.

Reply:
When you open a file with `FILE_APPEND_DATA`, it sets the file position to the end of the file (pos = end of file). However, when you also specify `OverwriteExisting = true`, it truncates the file to 0, which effectively sets the position to the beginning (pos = 0). This setup has no impact on the writing process itself, as it will always write to the end in both scenarios.

And, writing operations will always append to the end of the file in both cases. even if you set to `OpenFor::Writing` so at least one of them is wrong...

Reply:
A file can be written to more than once without being closed and reopened which means it won't get overwritten.

This is what the [Win32 API](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-writefile) says:

> To write to the end of file, specify both the Offset and OffsetHigh members of the OVERLAPPED structure as 0xFFFFFFFF. This is functionally equivalent to previously calling the CreateFile function to open hFile using FILE_APPEND_DATA access.

So according to that, `OpenFor::Writing` would not append to the file because `OpenFor::Writing` uses `GENERIC_WRITE` not `FILE_APPEND_DATA` on Windows.

Reply:
But you're using OverwriteExisting = true at the same  time which means the file will be cleared. So it still won't append.

And, because lpOverlapped is just `nullptr` https://github.com/UE4SS-RE/RE-UE4SS/blob/main/deps/first/File/src/FileType/WinFile.cpp#L99 and you're not using `FILE_FLAG_OVERLAPPED ` flag , effectively it will just append to the end everytime WriteFile is called.

And overlapped is something that allows you to write to the same file object asynchronously and that is why you need to specify the range for it to work (for example you can issue multiple write from multiple parts of memory), and also the meaning of setting offsets to 0xffffffff (to write to the end). And this has nothing to do with the normal behaviour of the `WriteFile`

The following should applies to the normal case:
> Considerations for working with synchronous file handles:
If **lpOverlapped is NULL**, the write operation starts at the **current file position** and WriteFile **does not return(( until the operation is complete, and the system updates the file pointer before WriteFile returns.
If lpOverlapped is not NULL, the write operation starts at the offset that is specified in the [OVERLAPPED](https://learn.microsoft.com/en-us/windows/desktop/api/minwinbase/ns-minwinbase-overlapped) structure and WriteFile does not return until the write operation is complete. The system updates the OVERLAPPED Internal and InternalHigh fields and the file pointer before WriteFile returns.

Current file position just increases everytime you write to the file.

Reply:
We don't want to append after the file is closed, that's not the goal of `NewFileDevice`.
We only want to append while the file is open.
This is what keeps the log file from growing indefinitely.

I don't see why this change is needed.
We clearly want to append, so we're using the append option.
It doesn't matter if it's redundant on Windows, it still logically makes sense to use append instead of write.
There's nothing wrong with this logic that I can see, and changing from append to write just makes it less clear what we're doing.

Reply:
Because for a file, write is enough to make it append to the end indefinitely. Append when open hints that you want to keep whatever in that file and append to the end but overwrite means you want to discard the old data..

Reply:
> Because for a file, write is enough to make it append to the end indefinitely. So using an append when open a file hints that you want to keep whatever in that file and append to the end of existing data but overwrite means you want to discard the old data..

Where is this definition of append ?
To me, appending to a file just means append something to the file, it doesn't imply anything about the lifetime of what you're appending to.

I can see how the name `OpenFor` could potentially be confusing though, and it can only be a good thing to create better documentation for what that param actually does and the fact that it operates fully independently of whether you're overwriting the file.

Reply:
Hmm.. idk maybe that's just me feels weird about it..?